src/coreclr/src/vm/domainfile.cpp

```diff
@@ -2366,6 +2346,164 @@ DomainAssembly::EnumMemoryRegions(CLRDataEnumMemoryFlags flags)
+    m_componentAssemblyCount = m_componentAssemblies->Size / sizeof(READYTORUN_COMPONENT_ASSEMBLIES_ENTRY);
+    
+    // Check if the current module's image has native manifest metadata, otherwise the current->GetNativeAssemblyImport() asserts.
+    m_manifestMetadata = peImage->GetNativeMDImport();
+
+    HENUMInternal assemblyEnum;
+    HRESULT hr = m_manifestMetadata->EnumAllInit(mdtAssemblyRef, &assemblyEnum);
+    mdAssemblyRef assemblyRef;
+    int assemblyIndex = 0;
+    while (m_manifestMetadata->EnumNext(&assemblyEnum, &assemblyRef))
+    {
+        LPCSTR assemblyName;
+        hr = m_manifestMetadata->GetAssemblyRefProps(assemblyRef, NULL, NULL, &assemblyName, NULL, NULL, NULL, NULL);
+        m_assemblySimpleNameToIndexMap.Add(new (amTracker.Track(pHeap->AllocMem((S_SIZE_T)sizeof(AssemblyNameIndex))))
+            AssemblyNameIndex(SString(SString::Ansi, assemblyName), assemblyIndex));
+        assemblyIndex++;
+    }
+}
+
+DomainCompositeImage::~DomainCompositeImage()
+{
+    CONTRACTL
+    {
+        DESTRUCTOR_CHECK;
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_ANY;
+        INJECT_FAULT(COMPlusThrowOM(););
+    }
+    CONTRACTL_END;
+}
+
+bool DomainCompositeImage::HasSimpleName(LPCUTF8 utf8SimpleName, uint32_t utf8Length) const
+{
+    return utf8Length == m_utf8SimpleNameLength && !memcmp(utf8SimpleName, m_utf8SimpleName, utf8Length);
+}
+
+bool DomainCompositeImage::TestAndClearRunEagerFixups()
+{
+    bool runEagerFixups = m_runEagerFixups;
+    m_runEagerFixups = false;
+    return runEagerFixups;
+}
+
+DomainCompositeImage *DomainCompositeImage::Open(
+    AppDomain *pDomain,
+    PEFile *peFile,
+    PEImage *peImage,
+    LPCUTF8 compositeImageName,
+    uint8_t compositeImageNameLength,
+    LoaderAllocator *loaderAllocator)
+{
+    READYTORUN_HEADER *header = peImage->GetLoadedLayout()->GetReadyToRunHeader();
+    if (header == NULL)
+    {
+        return NULL;
+    }
+    LoaderHeap *pHeap = loaderAllocator->GetHighFrequencyHeap();
+    AllocMemTracker amTracker;
+    DomainCompositeImage *result = new (amTracker.Track(pHeap->AllocMem((S_SIZE_T)sizeof(DomainCompositeImage))))
+        DomainCompositeImage(pDomain, peFile, peImage, header, compositeImageName, compositeImageNameLength, loaderAllocator, amTracker);
+    amTracker.SuppressRelease();
+    return result;
+}
+
+Assembly *DomainCompositeImage::LoadComponentAssembly(uint32_t rowid)
+{
+    AssemblySpec spec;
+    spec.InitializeSpec(TokenFromRid(rowid, mdtAssemblyRef), m_manifestMetadata, NULL);
+    return spec.LoadAssembly(FILE_LOADED);
+}
+
+PTR_READYTORUN_CORE_HEADER DomainCompositeImage::GetComponentAssemblyHeader(const SString& simpleName)
+{
+    simpleName.Normalize();
+    const AssemblyNameIndex *assemblyNameIndex = m_assemblySimpleNameToIndexMap.Lookup(simpleName);
+    if (assemblyNameIndex != NULL)
+    {
+        const byte *imageBase = (const byte *)m_peImage->GetLoadedLayout()->GetBase();
```

Comment:
I am surprised we even have a "byte" type. We use "BYTE" all over the place, so I'd stick with that.

Reply:
Thanks for spotting, I hit this before on one of my other change as the type only exists on Windows. Renamed in 7th commit.